lib/library.h

```diff
@@ -113,6 +113,7 @@ class CPPCHECKLIB Library {
     }
 
     /** get allocation id for function by name (deprecated, use other alloc) */
+    // cppcheck-suppress unusedFunction - only used in unit tests
```

Comment:
I would drop the mention of unit tests for obvious helper functions like this. Same for the other methods in this file.

I will review this in more detail later on though.